### DISCUSSION TITLE
Do we need to remove the silence segments before training the speaker verification ECAPA-TDNN?

### DISCUSSION THREAD

**Discussion by a43992899** [speechbrain/speechbrain#1653] [Q&A]
I am trying to fine tune the ECAPA-TDNN on my speaker recognition dataset, do I need to remove the silence segments (doing VAD) before training? Will it affects the final result?

**Comment by vic4code** [comment]
> I am trying to fine tune the ECAPA-TDNN on my speaker recognition dataset, do I need to remove the silence segments (doing VAD) before training? Will it affects the final result?

Is there silence in your test dataset as well? If so, what is the percentage of silence? I have trained ECAPA-TDN before on the instrument recognition task. when I trained the data after VAD, the results were very different. For my task, there was a lot of silence, so VAD was not appropriate for me.

**Comment by mravanelli** [comment]
@vic4code, If there is a lot of silence, I think adding a VAD should be appropriate, no?
As for the ECAPA-TDNN recipe, I don't see any step where we remove silences. We basically train the system using chunks of 3 seconds randomly sampled within each sentence. The assumption here is that the signal will mainly contain speech. I think small portions of noise are anyway tolerated.

**Comment by vic4code** [comment]
@mravanelli, perhaps the data we use is a little different from the common speech data. I conducted the ECAPA model on the Nsynth dataset containing instrumental sounds with a duration of 4 seconds for all data. However, about 10-15 % of the files start with a transient pulse of sound of about 0.1 seconds (silent in the left 3.9 secs). They are the sounds you get when you play a stringed instrument with the Pizzico or palm muting playing technique. So if I do VAD before training, there is a big mismatch that the model mainly learns to recognize the instrumental sounds without silence resulting in a poor F1-score.
